kentikapi/internal/api_payloads/device.go

```
@@ -423,14 +402,18 @@ func payloadToDevicePlan(p devicePlanPayload) (models.DevicePlan, error) {
 
 // labelIDPayload represents JSON ApplyLabels.LabelID payload as it is transmitted to KentikAPI.
 type labelIDPayload struct {
-	ID string `json:"id"`
+	ID int `json:"id"`
 }
 
 //nolint:revive // labelIDPayLoad doesn't need to be exported
 func LabelIDsToPayload(ids []models.ID) []labelIDPayload {
 	result := make([]labelIDPayload, 0, len(ids))
 	for _, id := range ids {
-		result = append(result, labelIDPayload{ID: id})
+		intID, err := strconv.Atoi(id)
+		if err != nil {
+			return []labelIDPayload{}
```

We should return error to the user instead of silently ignoring it.

---

Done